src/blockchains/evm/contracts/UniswapV3Factory.ts

```diff
@@ -21,26 +23,33 @@ export type PoolCreatedEvent = {
 
 @injectable()
 export class UniswapV3Factory {
+  @inject('Logger') logger!: Logger;
+  @inject('Settings') settings!: Settings;
+  @inject(BlockchainProviderRepository) blockchainProviderRepository!: BlockchainProviderRepository;
+
   protected ABI!: never;
 
-  readonly contractId: string = '';
```

Comment:
why readonly is removed? this is sigleton, if you change this in a middle of execution it will be changes everywhere, no?

Reply:
yeah, I need to change to repository/factory and it will have the readonly props instead of using setValues.